### PR TITLE
feat: add SAML app anonymous metadata and certificate APIs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add GET SAML app certificate and anonymous metadata APIs
1. GET /saml-applications/:id/metadata (resolves LOG-10118)
2. GET /saml-applications/:id/certificate (resolves LOG-10119, LOG-10525)

Also provide corresponding APIs for downloading metadata and certificate in file format.
1. GET /saml-applications/:id/metadata/download
2. GET /saml-applications/:id/certificate/download

## Testing

Covered by CI.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/saml-applications/routes/index.ts [hunk 1]
```diff
@@ -244,4 +250,28 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
       return next();
     }
   );
+
+  router.get(
+    '/saml-applications/:id/certificate',
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:255]: per offline discussion we need a download endpoint as well.

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 2]
```diff
@@ -91,9 +101,40 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
     });
   };
 
+  const getSamlApplicationMetadataByApplicationId = async (
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:104]:
```suggestion
  const getSamlIdPMetadataByApplicationId = async (
```

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 3]
```diff
@@ -91,9 +101,40 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
...
+        {
+          Location: buildSingleSignOnUrl(tenantEndpoint, id),
+          Binding: BindingType.Redirect,
+        },
+      ],
+    });
+
+    return {
+      metadata: idp.getMetadata(),
+      secretId,
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:130]: Just curious why do we need this?

File: packages/core/src/saml-applications/routes/index.ts [hunk 4]
```diff
@@ -252,4 +260,78 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
...
+
+      ctx.status = 200;
+      ctx.body = { certificate, fingerprints };
+
+      return next();
+    }
+  );
+
+  router.get(
+    '/saml-applications/:id/certificate/download',
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:289]: nit: How about /saml-applications/:id/certificate.pem ?

File: packages/core/src/saml-applications/routes/index.ts [hunk 5]
```diff
@@ -252,4 +260,78 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
...
+      ctx.set(
+        'Content-Disposition',
+        createContentDisposition(`saml-certificate-app-${id}-secret-${secretId}.pem`)
+      );
+
+      return next();
+    }
+  );
+
+  router.get(
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:314]: ditto: saml-applications/:id/metadata.xml ?

**Review by simeng-li — APPROVED** [review]
over all LGTM

File: packages/core/src/saml-applications/libraries/saml-applications.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/saml-applications/libraries/saml-applications.ts:130]: We use `secretId` as part of metadata/certificate's file name.

Comment (darcyYe) [reply packages/core/src/saml-applications/libraries/saml-applications.ts]: Removed, double checked certificates download from Azure/Okta, they are using general name like `okta.cert`.

**Pull request merged**